### PR TITLE
Allow PasswordAuthenticator to supply custom authenticators dynamically by clients

### PR DESCRIPTION
**Pull request by tejavadali**
This PR is inspired by the ongoing discussion in [here](https://github.com/gocql/gocql/pull/1573#issuecomment-898661961)

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -35,7 +35,8 @@ var (
 	}
 )
 
-func approve(authenticator string) bool {
+func approve(authenticator string, dynamicApprovedAuthenticators []string) bool {
+	approvedAuthenticators := append(dynamicApprovedAuthenticators, defaultApprovedAuthenticators...)
```
[the remark this answers is not in the document]
Comment (tejavadali) [reply conn.go:39]: fair point

[the remark this answers is not in the document]
Comment (tejavadali) [reply conn.go]: Addressed your comments, @martin-sucha  let me know if there is anything else I need to do.

[the remark this answers is not in the document]
Comment (martin-sucha) [reply conn.go]: Looks good to me. Thanks! Please [add yourself to AUTHORS file](https://github.com/gocql/gocql/blob/master/CONTRIBUTING.md#minimum-requirement-checklist) and we are good to go. Sorry I forgot to mention this during the first review.

[the remark this answers is not in the document]
Comment (tejavadali) [reply conn.go]: 👍  done :)

[the remark this answers is not in the document]
Comment (tejavadali) [reply conn.go]: @martin-sucha  let me know if need to do anything else. Thanks

**Comment by martin-sucha** [comment]
I've merged the changes, thanks! (I reworded the commit message and squashed commits, so GitHub did not mark this pull request as merged automatically)

**Comment by tejavadali** [comment]
@martin-sucha Sorry to bug you again, what is the release cadence we follow? When is this patch released and available to the general public to consume?

**Comment by martin-sucha** [comment]
It is merged to master, so it's available to everybody who updates to the latest version. Currently we don't tag releases, but we probably should, see https://github.com/gocql/gocql/issues/977

**Pull request closed without merging**